Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -78,45 +92,49 @@ void preparePathForEditing() {
             // Open a new path that the contents of the old path will be streamed into. We cannot re-use the same path as it is now closed
             // and cannot be opened again. We use the newPath variable because the factory was returning the same pointer for some strange
             // reason so this will force it to do otherwise.
-            FAIL_FAST_IF_FAILED(factory->CreatePathGeometry(&newPath));
-            FAIL_FAST_IF_FAILED(newPath->Open(&newSink));
-            FAIL_FAST_IF_FAILED(_impl.pathGeometry->Stream(newSink.Get()));
+            RETURN_IF_FAILED(factory->CreatePathGeometry(&newPath));
+            RETURN_IF_FAILED(newPath->Open(&newSink));
+            RETURN_IF_FAILED(PathGeometry()->Stream(newSink.Get()));
 
-            _impl.pathGeometry = newPath;
-            _impl.geometrySink = newSink;
+            PathGeometry() = newPath;
+            GeometrySink() = newSink;
 
             // Without a new figure being created, it's by default closed
-            _impl.isFigureClosed = true;
+            _impl.figureClosed = true;
         }
+        return S_OK;
     }
 
-    void closePath() {
-        if (_impl.geometrySink) {
-            endFigure(D2D1_FIGURE_END_OPEN);
-            _impl.geometrySink->Close();
-            _impl.geometrySink = nullptr;
+    void ClosePath() {
+        if (GeometrySink()) {
+            EndFigure(D2D1_FIGURE_END_OPEN);
+            GeometrySink()->Close();
+            GeometrySink() = nullptr;
         }
     }
 
-    void beginFigure() {
-        if (_impl.isFigureClosed) {
-            _impl.geometrySink->BeginFigure(_CGPointToD2D_F(_impl.currentPoint), D2D1_FIGURE_BEGIN_FILLED);
-            _impl.isFigureClosed = false;
+    void BeginFigure() {
+        if (_impl.figureClosed) {
+            GeometrySink()->BeginFigure(_CGPointToD2D_F(CurrentPoint()), D2D1_FIGURE_BEGIN_FILLED);
+            _impl.figureClosed = false;
         }
```

Comment:
This is getting completely out-of-hand.
We are supposed to introduce these sorts of things to *make the code clearer*.

`Impl` is a dumb data container, for all intents and most purposes. This isn't an exposure across a public interface. If every private implementation detail needs a protected-private-public interface, we'll find ourselves drowning in methods with identical names where a basic indirect and field access would work.

`Impl` is **literally** the struct we have to use because we can't put the members on the outer struct. It doesn't need to be treated like a separate entity.

---
In reply to: [85838765](https://github.com/Microsoft/WinObjC/pull/1255#discussion_r85838765) [](ancestors = 85838765)